lumen_web/src/event/target_1.rs

```diff
@@ -4,26 +4,27 @@
 //! end
 //! ```
 
+use liblumen_alloc::atom;
 use liblumen_alloc::erts::exception;
 use liblumen_alloc::erts::process::Process;
-use liblumen_alloc::erts::term::Term;
+use liblumen_alloc::erts::term::prelude::*;
 
 use lumen_runtime_macros::native_implemented_function;
 
-use crate::{error, event, ok};
+use crate::event;
 
 #[native_implemented_function(target/1)]
-fn native(process: &Process, event_term: Term) -> exception::Result {
+fn native(process: &Process, event_term: Term) -> exception::Result<Term> {
     let event = event::from_term(event_term)?;
 
     match event.target() {
         Some(event_target) => {
             let event_target_resource_reference = process.resource(Box::new(event_target))?;
 
             process
-                .tuple_from_slice(&[ok(), event_target_resource_reference])
+                .tuple_from_slice(&[atom!("ok"), event_target_resource_reference])
                 .map_err(|error| error.into())
         }
-        None => Ok(error()),
+        None => Ok(atom!("error")),
```

Comment:
Why is it better to having the string here than a function?  We know that LLVM+Rust hasn't been great about outlining and is better at deciding if it should inline or not.

Reply:
So this produces a `Term` value, and the previous `error` function just called `atom!("error")` itself. Using `atom!` directly here makes it clearer what is being constructed, and how. 

What we care about inlining (or not) is the construction of the atom itself, since that is where the real work takes place. The presence of the string will be lifted to a global, so duplication there won't have any impact on code size there.

In general though, I absolutely don't want to write code based on how the Rust code generator does things. 1.) That behavior can change at any time, and almost certainly will with each release, not to mention it changes with different compiler flags as it is; and 2.) We end up writing less readable code as a result of trying to force the compiler to generate code a specific way, which can break with seemingly unrelated changes elsewhere in the code base. I'll eat a performance hit or code size hit in some cases, knowing that we can work on that as a separate issue either by special casing things in egregious cases, or by upstreaming work to the Rust compiler or LLVM if we find systemic issues with code size on wasm32.